### ISSUE TITLE
I think I got everything set up right but keep get 502 bad gateway 

### ISSUE THREAD

**Issue by jlopez777** [YunoHost-Apps/wireguard_ynh#25]
### Describe the bug

Keep getting 502 Bad Gateway and not sure if its related to bug or configuration error on my side. 

### Context

Once I learned that wireguard has to be on root domain - I deleted all apps, updated my yunohost box and reinstalled wireguard. 

### Hardware - Recycled laptop, connected via ethernet

...*
- YunoHost version: Powered by YunoHost 4.2.6.1 (stable).

- I have access to my server: Through SSH | through the webadmin 

- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *no / yes*
  - 

### Steps to reproduce

- install wireguard
- click on wireguard tile launcher and get 502 badgatway. 

go to services and see that the wireguard_ui.service failed. I do a diagoniss and it recommens restarting it. 

I get success notification for restart, but still fails to start. Here are the logs. 
```
Jun 25 23:37:06 systemd[1]: Started WireGuard UI.
  - Jun 25 23:37:06 systemd[784]: wireguard_ui.service: Changing to the requested working directory failed: Permission denied
  - Jun 25 23:37:06 systemd[784]: wireguard_ui.service: Failed at step CHDIR spawning /opt/yunohost/wireguard/wireguard-ui: Permission denied
  - Jun 25 23:37:06 systemd[1]: wireguard_ui.service: Main process exited, code=exited, status=200/CHDIR
  - Jun 25 23:37:06 systemd[1]: wireguard_ui.service: Failed with result 'exit-code'.
  - Jun 25 23:38:55 systemd[1]: Started WireGuard UI.
  - Jun 25 23:38:55 systemd[1691]: wireguard_ui.service: Changing to the requested working directory failed: Permission denied
  - Jun 25 23:38:55 systemd[1691]: wireguard_ui.service: Failed at step CHDIR spawning /opt/yunohost/wireguard/wireguard-ui: Permission denied
  - Jun 25 23:38:55 systemd[1]: wireguard_ui.service: Main process exited, code=exited, status=200/CHDIR
  - Jun 25 23:38:55 systemd[1]: wireguard_ui.service: Failed with result 'exit-code'.
  - Jun 25 23:41:47 systemd[1]: Started WireGuard UI.
  - Jun 25 23:41:47 systemd[1860]: wireguard_ui.service: Changing to the requested working directory failed: Permission denied
  - Jun 25 23:41:47 systemd[1860]: wireguard_ui.service: Failed at step CHDIR spawning /opt/yunohost/wireguard/wireguard-ui: Permission denied
  - Jun 25 23:41:47 systemd[1]: wireguard_ui.service: Main process exited, code=exited, status=200/CHDIR
  - Jun 25 23:41:47 systemd[1]: wireguard_ui.service: Failed with result 'exit-code'.
  - Jun 26 00:07:49 systemd[1]: Started WireGuard UI.
  - Jun 26 00:07:49 systemd[4142]: wireguard_ui.service: Changing to the requested working directory failed: Permission denied
  - Jun 26 00:07:49 systemd[4142]: wireguard_ui.service: Failed at step CHDIR spawning /opt/yunohost/wireguard/wireguard-ui: Permission denied
  - Jun 26 00:07:49 systemd[1]: wireguard_ui.service: Main process exited, code=exited, status=200/CHDIR
  - Jun 26 00:07:49 systemd[1]: wireguard_ui.service: Failed with result 'exit-code'.
```
I'm lost bc I installed on defaul yunohost provided domain

I uncommented those 2 lines, 
I added port fwd on my router
and went into config and added ip tabels post up / down. 

### Expected behavior
I install and click on wireguard and be able to configure, oddly enough, when i had all my other apps, i could get to the wireguard ui. But not that I have nothing else installed, I get bad gateway. 

### Logs
```
 wg-quick@wg0.service - WireGuard via wg-quick(8) for wg0
   Loaded: loaded (/lib/systemd/system/wg-quick@.service; enabled; vendor preset: enabled)
   Active: active (exited) since Fri 2021-06-25 23:41:17 EDT; 32min ago
     Docs: man:wg-quick(8)
           man:wg(8)
           https://www.wireguard.com/
           https://www.wireguard.com/quickstart/
           https://git.zx2c4.com/wireguard-tools/about/src/man/wg-quick.8
           https://git.zx2c4.com/wireguard-tools/about/src/man/wg.8
  Process: 1814 ExecStart=/usr/bin/wg-quick up wg0 (code=exited, status=0/SUCCESS)
 Main PID: 1814 (code=exited, status=0/SUCCESS)

Jun 25 23:41:17 jsl777.nohost.me systemd[1]: Starting WireGuard via wg-quick(8) for wg0...
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: Warning: `/etc/wireguard/wg0.conf' is world accessible
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: [#] ip link add wg0 type wireguard
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: [#] wg setconf wg0 /dev/fd/63
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: [#] ip -4 address add 10.10.10.0/24 dev wg0
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: [#] ip link set mtu 1450 up dev wg0
Jun 25 23:41:17 jsl777.nohost.me wg-quick[1814]: [#] iptables -A FORWARD -i wg0 -j ACCEPT; iptables -A FORWARD -o wg0 -j ACCEPT; iptables -t nat -A POSTROUTING -o eth
Jun 25 23:41:17 jsl777.nohost.me systemd[1]: Started WireGuard via wg-quick(8) for wg0.
~
net.ipv4.conf.default.rp_filter = 1
net.ipv4.conf.all.rp_filter = 1
net.ipv4.tcp_syncookies = 1
net.ipv4.ip_forward = 1
net.ipv6.conf.all.forwarding = 1

-
 This file was generated using wireguard-ui (https://github.com/ngoduykhanh/wireguard-ui)
# Please don't modify it manually, otherwise your change might got replaced.

# Address updated at:     
# Private Key updated at: 
[Interface]
Address = xxxx (should my public ip be here? )
ListenPort = 8096
PrivateKey = <PRIVATE_KEY>
MTU = 1450
PostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE; ip6tables -A FORWARD -i %i -$
PostDown =iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE; ip6tables -D FORWARD -i %i $
```

**Comment by tituspijean** [comment]
Thank you for your report.

> Once I learned that wireguard has to be on root domain - I deleted all apps, updated my yunohost box and reinstalled wireguard.

Was the warning "WireGuard needs its own domain" in the installation form unclear?

> ```
> Jun 26 00:07:49 systemd[4142]: wireguard_ui.service: Changing to the requested working directory failed: Permission denied
> Jun 26 00:07:49 systemd[4142]: wireguard_ui.service: Failed at step CHDIR spawning /opt/yunohost/wireguard/wireguard-ui: > Permission denied
> ```

Can you share the result of `ls -la /opt/yunohost/wireguard`?

**Comment by tituspijean** [comment]
Issue found and fixed in #27. Feel free to reopen if needed.
